torch/nn/functional.py

```diff
@@ -2806,6 +2806,33 @@ def smooth_l1_loss(
     return torch._C._nn.smooth_l1_loss(expanded_input, expanded_target, _Reduction.get_enum(reduction), beta)
 
 
+def huber_loss(
+    input: Tensor,
+    target: Tensor,
+    reduction: str = 'mean',
+    beta: float = 1.0,
+) -> Tensor:
+    r"""Function that uses a squared term if the absolute
+    element-wise error falls below beta and a beta-scaled L1 term otherwise.
+
+    See :class:`~torch.nn.HuberLoss` for details.
+    """
+    if not torch.jit.is_scripting():
+        tens_ops = (input, target)
+        if any([type(t) is not Tensor for t in tens_ops]) and has_torch_function(tens_ops):
+            return handle_torch_function(
+                huber_loss, tens_ops, input, target,
+                reduction=reduction, beta=beta)
+    if not (target.size() == input.size()):
+        warnings.warn("Using a target size ({}) that is different to the input size ({}). "
```

Comment:
Aha, so here's the warning that's parallel to the C++ warning. Same question: do we really want to warn people if they do this?

Reply:
I think I'd rather have no warning, but that makes it inconsistent with other loss implementations.